src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -40,12 +40,13 @@ public static double toDouble(Object value) throws NumberFormatException {
      *
      * @param expenseIDAdd ExpenseID required for adding a new expense to the designated monthID MonthlyStorage Object.
      * @return ExpenseOD String message indicating success/fail add attempt.
+     * @throws EntityException thrown when the new expense input is invalid.
      */
     @Override
-    public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) {
+    public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) throws EntityException {
+        SessionStorage session = expenseIDAdd.getSession();
+        this.month = session.getMonthlyData(expenseIDAdd.getMonthID());
         try {
-            SessionStorage session = expenseIDAdd.getSession();
-            this.month = session.getMonthlyData(expenseIDAdd.getMonthID());
```

Comment:
@luluwuwu what do you think of this change. My reasoning can be seen in the commit message:

 Made addExpenseInMonth and addCategoryInMonth methods throw EntityException to fix the IntelliJ warning that only the interface was throwing but not any actual implementations. Also, I believe this is more correct since the EntityException caused by having the wrong monthID is caught in the view and shouldn't be caught here where the error message says it's because there's a category/expense with the same name.

Reply:
Yep, moving the getMonthlyData outside the try catch is what I did